### ISSUE TITLE
Story: Add Audio-Described Files to Video Content Type and Legacy Embedded Video Type

### ISSUE THREAD

**Issue by andyvanavery31** [NCIOCPL/cgov-digital-platform#4484]
### Specifications

**General**
- The addition of the audio-described file shall use a simple approach that does not change the video player itself
- The addition of the audio-described file shall assume that the video will no longer be a page-level content type
- We will not support DCEG Connect with this feature. No styling will be applied to an AD link on one of their videos if they add it.

**Drupal Specifications**
- The audio-described version of the video will be accessed through an external link
- The audio-described version of the video will be optional for videos
- This ticket will include the addition of the audio-described paragraph to the CGDP Video Content Type
- Developer Note: If the video does not have a caption, and it has an audio-described link then the FigCaption should not be created, and the text and link for the audio-described link will not be a description for the video

**Front-End Specifications**
- The video component shall allow users to access the audio-described version of the video
- If the video does not have an audio described version to go with it, then the designed interface to go to the audio-described version will not render
- The embedded video component update shall have the same user experience on the following embedding types on the Digital Platform:
  - Video Display: Large: No Title
  - Video Display: Large: Title
  - Video Display: Medium: No Title
  - Video Display: Medium: Title
  - Video Display: Small: No Title
  - Video Display: Small: Title
  - Alignement: None, Left, Center, and Right
- The embedded video component can be placed on the Body Content WYSIWYG on the following Content Pages:
  - Article Page
  - Biography
  - Blog Post
  - Cancer Center
  - Cancer Research List Page
  - Event
  - Press Release
- The Video Page will behave and look the same as the legacy embedded videos
- If the video has a caption, then the text and link to the audio-described link will appear in the caption
- If the video does not have a caption, then the audio-described link will appear alone
- The icon svg should be marked as hidden since the decorative icon does not add value for non sighted users.

**Analytics**
- No analytics required. This component is to ensure videos are accessible. A low Click Through Rate (CTR) will not change or remove this feature.

### DoR Tasks

- [x] Content Model Drafted
- [x] Content Model Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Content Model Approved
  - [x] Developer Approval
  - [x] IA Approval
- [x] Acceptance Criteria Drafted
- [x] Scenarios created for YAML
- [x] Acceptance Criteria Reviewed
  - [x] Developer Review
  - [x] IA Review
- [x] Acceptance Criteria Approved
  - [x] Developer Approval
  - [x] IA Approval

### DoD Tasks

- [ ] Governance Drafted
- [ ] PR Review (see PR for review and comments)
- [ ] Client review with demo

### Technical Notes

**Content Model**
- [Video Content Type](https://github.com/NCIOCPL/cgov-digital-platform/wiki/Media-Type-Video)

**Scenarios/Comps**
- On an example Article page for test content, use the [Embedded Video Sizes](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8256-50059&node-type=frame&t=Rbq3Wb7vBFXurukc-0) and [Legacy Video Sizes](https://www.figma.com/design/GCOniv8OTyj9OC1udKwo6S/Mini-Landing-Pages?node-id=8886-13735&node-type=text&m=dev) comp as a reference:
  - Page URL: Something similar to /test/embedded-video-sizes/
  - Page Title: Test Page for Embedded Video Sizes
  - Components:
    - Add every combination of video size (3 - large, medium, small), with or without title, alignment (4 - None, Left, Center, Right), and with or without caption. All videos placed on the page will need an external link to be added to the video content type. This should be 48 videos. Around each video, add 3 lines of lorem ipsum above and below the video to test the wrapping.
- **NOTE**: Legacy styling will apply the external link icon to the end of the link

### Acceptance Criteria

**CMS Experience - Adding AD Link to Video Media Type**
```gherkin
Scenario: A content editor adds or edits a video media type on Drupal to add an audio-described link
  Given the user is logged into the CMS as an editor
    And the user is adding or editing a video media type
  When the user is adding or editing a video media type
  Then the user will see the fields for the video media type
    And the "Audio-Described Video Link" field will be added after the Caption field
    And the Audio-Described Video Link field will be optional
    And it will be a text entry field
    And it will have the tooltip text: "Used to provide a link to the audio-described file. Enter the full URL starting with https://"
    And all other fields will remain unchanged
```

**CMS Experience - Adding an Embedded Video to a Page Using the WYSIWYG**
```gherkin
Scenario: A content editor adds an embedded video to a page using the WYSIWYG
  Given the user is logged into the CMS as an editor
    And the user is adding an embedded video to a page using the WYSIWYG
  When the user is adding an embedded video to a page using the WYSIWYG
  Then the experience does not change whether the video has an audio-described link or not
```

**Front-End Experience**
```gherkin
Scenario: A user views a video with and audio-described link on an applicable page on the platform
  Given the audio-described link has been added to the video media type
    And the video with an audio-described link has been added to an applicable page on the platform
    And the applicable pages for Legacy Embedded WYSIWYG are, Article Page, Biography, Blog Post, Cancer Center, Cancer Research List Page, Event, and Press Release
  When the video is embedded at any size
    And it has or does not have a title
    And it has a caption
  Then the audio-described link will appear below the caption with the text, "<AD Icon> Play the audio-described version of this video."
    And it will match the comp
  When the video is embedded at any size
    And it has or does not have a title
    And it does not have a caption
  Then the audio-described link will appear below the video with the text, "<AD Icon> Play the audio-described version of this video."
  When the video is on a Spanish page
    And the video is embedded at any size
    And it has or does not have a title
    And it has a caption
  Then the audio-described link will appear below the caption with the text, "<AD Icon> Escuchar la descripción del video."
  When the video is on a Spanish Page
    And the video is embedded at any size
    And it has or does not have a title
    And it does not have a caption
  Then the audio-described link will appear below the video with the text, "<AD Icon> Escuchar la descripción del video."
  When the video does not have an audio-described link
    And it is embedded
    And it has or does not have a title
    And it has or does not have a caption
  Then it appears the same as it would before the change from this ticket
  When the user clicks on the link for the audio-described link
  Then they will be redirected to the external link for the audio-described video
```

### Related Tickets
* META: #4318

**Comment by blilianyu** [comment]
@dev-rana-publicis @andyvanavery31 Here is the Audio Describe icon.
![audio-description](https://github.com/user-attachments/assets/819521cc-e388-4e34-8e76-a9ba2952118a)

**Comment by dev-rana-publicis** [comment]
After speaking with Zoltan Hawryluk, who is a PS A11y specialist, we found that having the audio-described link inside of the figcaption, even though figcaptions are not being read by screenreaders on some browsers, is not the best practice nor does it abide by MDN documentation. To fix this, we will have to put the audio-described link outside of the figure element all together. This led to some pretty messy css in our already messy legacy stylesheet but I’ve left some dev notes and comments on the PR. To account for browsers that aren’t reading the figcaption, we are adding an aria-labelledby on the figure’s video, and an id on the figcaption. 
@andyvanavery31 @bryanpizzillo @olitharp-nci
